week2/homework/src/actions.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const fs = require('fs');
+
+const ENCODING = 'utf8';
+const PATH = `${__dirname}/todoList.json`;
+
+const actions = {
+  list: () => {
+    fs.readFile(PATH, ENCODING, (err, todoList) => {
+      if (err) {
+        if (err.code === 'ENOENT') console.log('no data found');
+        throw err;
+      }
+      console.log(`\nTodo list:\n${todoList}`);
+    });
+  },
+  add: todoItem => {
+    fs.readFile(PATH, ENCODING, (err, todoList) => {
+      if (err) {
+        if (err.code === 'ENOENT') console.log('no data found');
+        throw err;
+      }
+      todoList = JSON.parse(todoList);
+      todoList.push(todoItem);
+      fs.writeFile(PATH, JSON.stringify(todoList), err => {
+        if (err) throw err;
+      });
+      console.log(`'${todoItem}' has been added to the Todo list.`);
+    });
+  },
+  remove: index => {
+    fs.readFile(PATH, ENCODING, (err, todoList) => {
+      if (err) {
+        if (err.code === 'ENOENT') console.log('no data found');
+        throw err;
+      }
+      todoList = JSON.parse(todoList);
+      if (index > 0 && index <= todoList.length) {
+        todoList.splice(index - 1, 1);
+        console.log('Todo item has been removed');
+        fs.writeFile(PATH, JSON.stringify(todoList), err => {
+          if (err) throw err;
+        });
+      } else {
+        console.log('Please insert a valid number.');
+      }
+    });
+  },
+  reset: () => {
+    fs.writeFile(PATH, JSON.stringify([]), err => {
+      if (err) throw err;
+    });
+  },
+  update: (index, updatedItem) => {
+    fs.readFile(PATH, ENCODING, (err, todoList) => {
+      if (err) {
+        if (err.code === 'ENOENT') console.log('no data found');
+        throw err;
+      }
+      todoList = JSON.parse(todoList);
+      if (index > 0 && index <= todoList.length) {
+        todoList.splice(index - 1, 1, updatedItem);
+        console.log('Todo item has been modified');
```

Comment:
This also has to be moved to inside the callback of `fs.writeFile`.

Reply:
Thanks... done 👍